Hw4_Module/Hw4_Module/Program.cs

```diff
@@ -0,0 +1,258 @@
+            int columns = itemsArray.Length / rows;
+            for (int i = 0; i < rows; i++)
+            {
+                for (int j = 0; j < columns; j++)
+                {
+                    string curentItem = itemsArray[i, j];
+                    if (itemsArray[i, j] == null) 
+                    {
+                        continue;
+                    }
+                    else if ((curentItem.ToUpper() == item.ToUpper()) && indicateAddOrRemoveOrChangeStatus == "Search")
+                    {
+                        return true;
+                    }
+                    else if ((curentItem.ToUpper() == item.ToUpper()) && (indicateAddOrRemoveOrChangeStatus == "Remove"))
+                    {
+                        itemsArray[i, j] = null;
+                        itemsArray[i + 1, j] = null;
+                        return true;
+                    }
+                    else if ((curentItem.ToUpper() == item.ToUpper()) && indicateAddOrRemoveOrChangeStatus == "ChangeStatus") 
+                    {
+                        itemsArray[i + 1, j] = status;
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        public static void ShowItemsFromArray(ref string[,] itemsArray, string status) 
+        {
+            int rows = itemsArray.GetUpperBound(0) + 1;
+            int columns = itemsArray.Length / rows;
+            string[] SymbolInSecondLineInArray = new string[status.Length];
+            for (int i = 0; i < rows-1; i++)
+            {
+                for (int j = 0; j < columns; j++)
+                {
+                    if ((itemsArray[i, j] == null) && (itemsArray[i + 1, j] == null))
+                    {
+                        continue;
+                    }
+                    else if ( (itemsArray[i+1, j].Length > 1) && (status=="1")) // status = 1
+                    {
+                        SymbolInSecondLineInArray = itemsArray[i + 1, j].Split(' ');
+                        string statusFromArray = SymbolInSecondLineInArray[0];
+                        if (statusFromArray == status) 
+                        {
+                            Console.WriteLine($"Item - {itemsArray[i, j]}, status - {itemsArray[i + 1, j]}");
+                        }
+                    }
+                    else if (itemsArray[i + 1, j] == status) //status = 0
+                    {
+                        Console.WriteLine($"Item - {itemsArray[i, j]}, status - {itemsArray[i + 1, j]}");
+                    }
+                    else if (status=="") //status = ""
+                    {
+                        Console.WriteLine($"Item - {itemsArray[i, j]}, status - {itemsArray[i + 1, j]}");
+                    }
+                }
+            }
+        }
+
+        public static bool SearchFreeSpaceInArray(ref string[,] itemsArray, string item, string status="Search")
+        {
+            int counterFreeSpace = 0;
```

Comment:
Same, you are searching for an empty position in the array, and adding items. It should be a separate functionality.